src/classes/CountertopWorker.js

```diff
@@ -0,0 +1,132 @@
+import { IAppliance } from '@tvkitchen/base-interfaces'
+import { Payload } from '@tvkitchen/base-classes'
+import { applianceEvents } from '@tvkitchen/base-constants'
+import { consoleLogger } from '../tools/loggers'
+import kafka from '../tools/kafka'
+import { getStreamTopic } from '../tools/utils/countertop'
+
+/**
+ * A CountertopWorker (aka Line Cook) is responsible for monitoring specific data streams
+ * and processing those streams using TV Kitchen Appliances to create new data.
+ *
+ * CountertopWorker instances consume Payloads from Kafka and pass them to their Appliances.
+ * CountertopWorker instances listen for emitted Payloads from appliances and process them.
+ */
+class CountertopWorker {
+	logger = null
+
+	consumer = null
+
+	producer = null
+
+	admin = null
+
+	appliance = null
+
+	stream = null
+
+	/**
+	 * Create a new CountertopWorker.
+	 *
+	 * @param  {Class}            Appliance     The class for the appliance that this worker will use.
+	 * @param  {CountertopStream} stream        A stream that ends in this worker's station.
+	 * @param  {Object}           configuration Configuration settings for the Appliance.
+	 */
+	constructor(
+		Appliance,
+		applianceSettings = {},
+		{
+			stream,
```

Comment:
Is passing stream here appropriate?

It is used to decide which kafka topics to track / what the kafka group should be.  Note: nobody knows that workers use Kafka this way, which is good (vs, for instance, passing in things like `kafkaStream` or `kafkaGroup`)

Tl;DR it feels odd for the Worker to have to know its context, but ultimately it does need to know what data streams to read and write from, and all of that is contained by the CountertopStream that is passed here.

Reply:
Seems fine.